torch/fx/experimental/Partitioner.py

```diff
@@ -156,13 +232,92 @@ def find_single_partition(self) -> None:
+           The placeholder nodes (placeholder) and constant nodes (get_attr) are skipped.
+           A placeholder (placeholder) or a constant (get_attr)
+           is added into a partition when it is a input node for a op node.
+           For one op node to another, check if a op node and its input nodes
+           can fit into the current partition.
+           If the current partition is full, create a new one
+           and continue traversing op nodes.
+           For orphan nodes (placeholder or constants are never used),
+           a new partition will create for each of them.
+           Then through self.combine_partition_based_on_size(),
+           partitions will be combined to keep
+           as less partitions as possible
+        """
+        self.users_mapping = _get_users_for_all_nodes(self.graph_module)
+        self.args_mapping = _get_args_for_all_nodes(self.users_mapping)
+        partition = self.create_partition()
+        # track the used mem for the current partition
+        used_mem = 0
+        for node in self.graph_module.graph.nodes:
+            if node.op in {'call_module', 'call_method', 'call_function'}:
+                # Find all its input nodes
+                input_nodes = self.args_mapping[node]
+                # Calculate total size of input nodes
+                total_size_of_input_nodes = 0
+                for n in input_nodes:
+                    # Make sure this node hasn't been in this partition yet
+                    if n not in partition.nodes:
+                        assert hasattr(n, 'size')
+                        total_size_of_input_nodes += n.size.output_size
+                # Don't forget add the size of the current node
+                total_size_of_input_nodes += node.size.total_size
+                if used_mem + total_size_of_input_nodes > available_mem:
+                    # The current node cannot fit into the current partition
+                    self.partition_to_used_mem[partition] = used_mem
+                    partition = self.create_partition()
+                    used_mem = 0
+                    # The current node may be too large to fit into a whole new partition
+                    if total_size_of_input_nodes > available_mem:
+                        raise RuntimeError(node.name + 'is too large to fit into a device')
+                # Add the current node into the current partition
+                partition.add_node(node)
+                # Add all input nodes if they are placeholders or constants
+                for n in input_nodes:
+                    if (n not in partition.nodes) and (n.op in {'placeholder', 'get_attr'}):
+                        partition.add_node(n)
+                used_mem = used_mem + total_size_of_input_nodes
+        # Don't forget to update used mem mapping for the last partition
+        self.partition_to_used_mem[partition] = used_mem
+        # Find all orphans (placeholders or constants are never used)
```

Comment:
Are these nodes dead code? When would you encounter these?

Reply:
For example, 
class TestModule:
    def forward(sef, a, b, c)
        e = torch.nn.rand(1)
        return a + b, c, e
c  and e are orphan nodes

Reply:
Don't know if I fully understand. Are you only considering the first output of the Graph?

Reply:
I think the case I try to show is that how to deal with some placeholders and constants that don't do anything but are returned.

Reply:
A better example is:
        def forward(self, a):
            b = torch.rand(4)
            return b
'a' and 'b' don't do anything here. They are 'orphans'.
Since those placeholders and constants are never used, when we use split_module, they are not in the submodules. So I think we could ignore them.